v03_pipeline/lib/tasks/update_variant_annotations_table_with_new_samples.py

```diff
@@ -161,41 +107,18 @@ def complete(self) -> bool:
         )
 
     def update_table(self, ht: hl.Table) -> hl.Table:
-        callset_hts = [
-            hl.read_matrix_table(
-                remapped_and_subsetted_callset_path(
-                    self.reference_genome,
-                    self.dataset_type,
-                    callset_path,
-                    project_guid,
-                ),
-            ).rows()
-            for (callset_path, project_guid, _, _) in callset_project_pairs(
-                self.callset_paths,
-                self.project_guids,
-                self.project_remap_paths,
-                self.project_pedigree_paths,
-            )
-        ]
-
-        # Drop any fields potentially unshared/unused by the annotations.
-        for i, callset_ht in enumerate(callset_hts):
-            for row_field in self.dataset_type.optional_row_fields:
-                if hasattr(callset_ht, row_field):
-                    callset_hts[i] = callset_ht.drop(row_field)
-
-        callset_ht = functools.reduce(
-            (lambda ht1, ht2: ht1.union(ht2, unify=True)),
-            callset_hts,
+        new_variants_ht = hl.read_table(
+            new_variants_table_path(
+                self.reference_genome,
+                self.dataset_type,
+            ),
         )
-        callset_ht = callset_ht.distinct()
 
-        # 1) Get new rows and annotate with vep
+        # Get new rows and annotate with vep.
```

Comment:
I kept vep in `UpdateVariantAnnotationsTableWithNewSamplesTask` because I want to slot the new clingen registering step in between `WriteNewVariantsTable` and `UpdateVariantAnnotationsTableWithNewSamplesTask` - which means vep would run only if registering succeeds.

Reply:
I guess VEP really should be in its own task, so maybe now's the time to do that

Reply:
yeah... there's definitely some uncertainty here.  I think at this point we have three separate things that we do for the new variants: 1) clingen register 2) VEP 3) formatting annotations.  We could put all three in one task, or split them apart.

Reply:
Now, all the new variant stuff is in `WriteNewVariantsTableTask`!

Reply:
🤟